libs/ledger-live-common/src/hw/index.ts

```diff
@@ -107,7 +112,45 @@ export const open = (
   for (let i = 0; i < modules.length; i++) {
     const m = modules[i];
     const p = m.open(deviceId, timeoutMs, context);
-    if (p) return p;
+    if (p) {
+      trace({
+        type: LOG_TYPE,
+        message: `Found a matching Transport: ${m.id}`,
+        context,
+        data: { timeoutMs },
+      });
+
+      if (!timeoutMs) {
+        return p;
+      }
+
+      let timer: ReturnType<typeof setTimeout> | null = null;
+
+      // Throws CantOpenDevice on timeout, otherwise returns the transport.
+      // Important: with Javascript Promise, when one promise finishes,
+      // the other will still continue, even if its return value will be discarded.
+      return Promise.race([
+        p.then(transport => {
```

Comment:
have you tested the (possible ? idk) case where `p` throws an error because the timeout implemented in `open()` times out (and thus throws an error) before the `setTimeout` called here ?

Reply:
If the Promise `p` throws an error because of a timeout before this `setTimeout`, it is fine because the `Promise.race` will only reject at the 1st rejection/error (it's not going to/cannot reject twice because the 2 threw an error each)

I can look to test this case, but not sure